src/comm/TcpAcceptor.cc

```diff
@@ -323,10 +324,14 @@ Comm::TcpAcceptor::notify(const Comm::Flag flag, const Comm::ConnectionPointer &
         params.xaction = new MasterXaction(XactionInitiator::initClient);
         params.xaction->squidPort = listenPort_;
         params.fd = conn->fd;
+        conn->enterOrphanage();
```

Comment:
This line looks wrong because 'conn' is the _listening_ connection; newConnDetails is the connection passed here from acceptOne() should be the subject of 'orphanage'.

Reply:
Fixed in b4e50a8